morse_decoder.rb

```diff
@@ -0,0 +1,37 @@
+# morse dictionary
+@morse_dictionary = { '.-' => 'A', '-...' => 'B', '-.-.' => 'C', '-..' => 'D',
+                      '.' => 'E', '..-.' => 'F', '--.' => 'G', '....' => 'H', '..' => 'I', '.---' => 'J', '-.-' => 'K',
+                      '.-..' => 'L', '--' => 'M', '-.' => 'N', '---' => 'O', '.--.' => 'P', '--.-' => 'Q',
+                      '.-.' => 'R', '...' => 'S', '-' => 'T', '..-' => 'U', '...-' => 'V', '.--' => 'W',
+                      '-..-' => 'X', '-.--' => 'Y', '--..' => 'Z' }
+
+def decode_char(str)
+  @morse_dictionary[str]
+end
+puts decode_char('-...')
+
+def decode_word(str)
+  str.split.map { |char| decode_char(char) }.join
+end
+puts decode_word('-... --- -..-')
+
+def decode(morse_str)
+  word = ''
+  final_text = ''
+
+  morse_str.split('   ').each do |i|
+    i.split.each do |r|
+      word.concat(@morse_dictionary[r])
+    end
+    final_text.concat("#{word} ")
+    word = ''
+  end
+  final_text
+end
```

Comment:
- [ ] Kindly be noticed that the `decode` method should reuse the `#decode_word` method to translate a single encoded-word into a word (DRY rule), so it would be better to make it like this.

```ruby
def decode(morse_str)
  morse_str.split('   ').map { |word| decode_word(word) }.join(' ')
end
```